lib/beautiful/gtk.lua

```diff
@@ -0,0 +1,178 @@
+---------------------------------------------------------------------------
+--- Querying current GTK+ 3 theme via GtkStyleContext.
+--
+-- @author Yauheni Kirylau &lt;yawghen@gmail.com&gt;
+-- @copyright 2016-2017 Yauheni Kirylau
+-- @module beautiful.gtk
+---------------------------------------------------------------------------
+local gears_debug = require("gears.debug")
+local unpack = unpack or table.unpack -- luacheck: globals unpack (compatibility with Lua 5.1)
+
+local gtk = {
+  cached_theme_variables = nil
+}
+
+
+local function convert_gtk_color_to_hex(gtk_color)
+  local rgb = gtk_color:to_string()
+  local m = rgb:gmatch("[0-9]+")
+  return m and (
+    "#" .. string.format("%2.2x", m()) .. string.format("%2.2x", m()) .. string.format("%2.2x", m())
+  )
+end
+
+
+local function lookup_gtk_color_to_hex(_style_context, color_name)
+  local color = _style_context:lookup_color(color_name)
+  if not color then
+      return nil
+  end
+  local result = convert_gtk_color_to_hex(color)
+  --color:free()
+  return result
+end
+
+local function get_gtk_color_property_to_hex(_style_context, property_name, state)
+    local property = _style_context:get_property(property_name, state)
+    if not property then
+      return nil
+    end
+    local result = convert_gtk_color_to_hex(property.value)
+    property:unset()
+    return result
+end
+
+
+--- Get GTK+3 theme variables from GtkStyleContext
+-- @treturn table subj
+function gtk.get_theme_variables()
+  if gtk.cached_theme_variables then
+    return gtk.cached_theme_variables
+  end
+
+  local result = {}
+  local lgi = require('lgi')
+  local Gtk = lgi.Gtk
+  local window
+  pcall(function()
+    window = Gtk.Window{
+      --on_destroy = Gtk.main_quit,
+    }
+  end)
+  if not window then
+    gears_debug.print_warning("Seems like GTK+3 is not installed or theme is not set correctly.")
+    return nil
+  end
+  --window:set_override_redirect(true)
+  local style_context = window:get_style_context()
+
+  local font = style_context:get_font(Gtk.StateFlags.NORMAL)
```

Comment:
This function is deprecated: https://developer.gnome.org/gtk3/stable/GtkStyleContext.html#gtk-style-context-get-font
(And no, I don't understand the replacement nor how to use it)